### PR TITLE
Feature/physics

### PR DESCRIPTION
**Pull request by Gorphi**

### CODE REVIEW

**Comment by Gorphi** [comment]
CatroidCheck: https://jenkins.catrob.at/view/LeeroyCatroid/job/CatroidCheck/63/
UnitTests: https://jenkins.catrob.at/view/LeeroyCatroid/job/CatroidSinglePackageEmulatorTest/80/
UiTests: https://jenkins.catrob.at/view/LeeroyCatroid/job/CatroidSinglePackageEmulatorTest/81/

File: catroid/res/values/arrays.xml [hunk 1]
```diff
@@ -154,4 +154,12 @@
         <item>@string/phiro_sensor_bottom_right</item>
     </string-array>
 
+    <!-- Physicsobjecttype -->
+    <string-array name="physics_object_types">
+        <item>@string/brick_set_physics_object_type_dynamic</item>
+        <item>@string/brick_set_physics_object_type_fixed</item>
+        <item>@string/brick_set_physics_object_type_no</item>
```
Comment (anddman) [inline catroid/res/values/arrays.xml]: *_none*?

File: catroid/src/org/catrobat/catroid/ProjectManager.java [hunk 2]
```diff
@@ -279,30 +279,23 @@ public boolean initializeDefaultProject(Context context) {
...
-			currentScript = null;
-			return true;
-		} catch (IOException ioException) {
-			Log.e(TAG, "Cannot initialize default project.", ioException);
-			Utils.showErrorDialog(context, R.string.error_load_project);
-			return false;
-		}
-	}
-
 	public void initializeNewProject(String projectName, Context context, boolean empty, boolean drone, boolean landscapeMode)
```
Comment (anddman) [inline catroid/src/org/catrobat/catroid/ProjectManager.java]: replace drone boolean parameter with ProjectCreatorType enum

File: catroid/src/org/catrobat/catroid/common/DefaultProjectHandler.java [hunk 3]
```diff
@@ -25,50 +25,45 @@
 import android.content.Context;
 import android.util.Log;
 
+import org.catrobat.catroid.BuildConfig;
 import org.catrobat.catroid.ProjectManager;
-import org.catrobat.catroid.R;
+import org.catrobat.catroid.common.standardprojectcreators.DefaultProjectCreator;
+import org.catrobat.catroid.common.standardprojectcreators.DefaultProjectCreatorDefault;
+import org.catrobat.catroid.common.standardprojectcreators.DefaultProjectCreatorDrone;
+import org.catrobat.catroid.common.standardprojectcreators.DefaultProjectCreatorPhysics;
```
Comment (anddman) [inline catroid/src/org/catrobat/catroid/common/DefaultProjectHandler.java]: refactor standardprojectcreators to defaultprojectcreators

File: catroid/src/org/catrobat/catroid/common/DefaultProjectHandler.java [hunk 4]
```diff
@@ -566,9 +111,21 @@ public static Project createAndSaveEmptyProject(String projectName, Context cont
...
-		int returnValue = (int) (value * backgroundImageScaleFactor);
-		int differenceToNextFive = returnValue % 5;
-		return returnValue - differenceToNextFive;
+	public void setDefaultProjectCreator(ProjectCreatorType type) {
+		switch (type) {
+			case PROJECT_CREATOR_DEFAULT:
+				defaultProjectCreator = new DefaultProjectCreatorDefault();
+				break;
+			case PROJECT_CREATOR_DRONE:
+				if (BuildConfig.FEATURE_PARROT_AR_DRONE_ENABLED) {
```
Comment (anddman) [inline catroid/src/org/catrobat/catroid/common/DefaultProjectHandler.java]: redundant?

File: catroid/res/values/arrays.xml [hunk 5]
```diff
@@ -154,4 +154,12 @@
         <item>@string/phiro_sensor_bottom_right</item>
     </string-array>
 
+    <!-- Physicsobjecttype -->
```
Comment (Gorphi) [inline catroid/res/values/arrays.xml]: Change To "Physics object type spinner"

File: catroid/res/values/strings.xml [hunk 6]
```diff
@@ -554,6 +571,9 @@
         <item quantity="other">"seconds "</item>
     </plurals>
 
+	<string name="brick_collision_receive">When physical collision with</string>
+	<string name="collision_with_anything">anything</string>
```
Comment (Gorphi) [inline catroid/res/values/strings.xml]: remove one tab

File: catroid/res/values/strings.xml [hunk 7]
```diff
@@ -605,6 +626,7 @@
     <string name="default_drone_project_name">My first Drone program</string>
     <string name="default_drone_project_sprites_takeoff">Take off</string>
 
+
```
Comment (Gorphi) [inline catroid/res/values/strings.xml]: remove newline

File: catroid/src/org/catrobat/catroid/common/DefaultProjectHandler.java [hunk 8]
```diff
@@ -566,9 +111,21 @@ public static Project createAndSaveEmptyProject(String projectName, Context cont
...
+					defaultProjectCreator = new DefaultProjectCreatorDrone();
+				} else {
+					defaultProjectCreator = new DefaultProjectCreatorDefault();
+				}
+				break;
+			case PROJECT_CREATOR_PHYSICS:
+				defaultProjectCreator = new DefaultProjectCreatorPhysics();
+				break;
+		}
```
Comment (Gorphi) [inline catroid/src/org/catrobat/catroid/common/DefaultProjectHandler.java]: replace strategy pattern if not needed.

File: catroid/src/org/catrobat/catroid/common/LookData.java [hunk 9]
```diff
@@ -130,6 +130,9 @@ public Pixmap getPixmap() {
 				if (gdxRuntimeException.getMessage().startsWith("Couldn't load file:")) {
 					pixmap = new Pixmap(1, 1, Pixmap.Format.Alpha);
 				}
+			} catch (NullPointerException nullPointerException) {
+				Log.d(TAG, "gdx.files throws nullpointer");
```
Comment (Gorphi) [inline catroid/src/org/catrobat/catroid/common/LookData.java]: change to Log.e(...)

File: catroid/src/org/catrobat/catroid/content/Look.java [hunk 10]
```diff
@@ -45,7 +45,7 @@
 public class Look extends Image {
 	private static final float DEGREE_UI_OFFSET = 90.0f;
 	private static ArrayList<Action> actionsToRestart = new ArrayList<Action>();
-	public boolean visible = true;
+	protected boolean lookVisible = true;
```
Comment (Gorphi) [inline catroid/src/org/catrobat/catroid/content/Look.java]: private should also be sufficient

File: catroid/src/org/catrobat/catroid/content/Project.java [hunk 11]
```diff
@@ -160,9 +164,15 @@ public XmlHeader getXmlHeader() {
 
 	public int getRequiredResources() {
 		int resources = Brick.NO_RESOURCES;
+		ActionFactory physicsActionFactory = new ActionPhysicsFactory();
```
Comment (Gorphi) [inline catroid/src/org/catrobat/catroid/content/Project.java]: create the factory just if required

File: catroid/src/org/catrobat/catroid/content/Sprite.java [hunk 12]
```diff
@@ -249,9 +258,22 @@ private void putBroadcastSequenceAction(String broadcastMessage, SequenceAction
...
+	}
+
 	@Override
 	public Sprite clone() {
 		final Sprite cloneSprite = new Sprite();
+		cloneSprite(cloneSprite);
+		return cloneSprite;
+	}
+
+	protected void cloneSprite(final Sprite cloneSprite) {
```
Comment (Gorphi) [inline catroid/src/org/catrobat/catroid/content/Sprite.java]: Why is that a outsourced to a different method?

File: catroid/src/org/catrobat/catroid/content/Sprite.java [hunk 13]
```diff
@@ -529,6 +555,27 @@ public String toString() {
 		return name;
 	}
 
+	public void rename(String newSpriteName) {
+		if ((getRequiredResources() & Brick.PHYSIC) > 0) {
+			List<Sprite> spriteList = ProjectManager.getInstance().getCurrentProject().getSpriteList();
+			for (Sprite currentSprite : spriteList) {
+				if ((currentSprite.getRequiredResources() & Brick.PHYSIC) > 0) {
+					currentSprite.updateCollisionBroadcastMessages(getName(), newSpriteName);
```
Comment (Gorphi) [inline catroid/src/org/catrobat/catroid/content/Sprite.java]: add early termination.

File: catroid/src/org/catrobat/catroid/content/Sprite.java [hunk 14]
```diff
@@ -529,6 +555,27 @@ public String toString() {
...
+				}
+			}
+		}
+		setName(newSpriteName);
+	}
+
+	public void updateCollisionBroadcastMessages(String oldCollisionObjectIdentifier, String newCollisionObjectIdentifier) {
```
Comment (Gorphi) [inline catroid/src/org/catrobat/catroid/content/Sprite.java]: public required?

File: catroid/src/org/catrobat/catroid/content/bricks/BroadcastBrick.java [hunk 15]
```diff
@@ -201,6 +202,11 @@ protected boolean handleOkButton() {
 					return false;
 				}
 
+				if (newMessage.contains(PhysicsCollision.COLLISION_MESSAGE_CONNECTOR)) {
+					Utils.showErrorDialog(getActivity(), R.string.brick_broadcast_invalid_symbol);
+					return false;
```
Comment (Gorphi) [inline catroid/src/org/catrobat/catroid/content/bricks/BroadcastBrick.java]: call to dismiss() missing?

File: catroid/src/org/catrobat/catroid/content/bricks/BroadcastReceiverBrick.java [hunk 16]
```diff
@@ -205,6 +207,11 @@ protected boolean handleOkButton() {
 					return false;
 				}
 
+				if (newMessage.contains(PhysicsCollision.COLLISION_MESSAGE_CONNECTOR)) {
+					Utils.showErrorDialog(getActivity(), R.string.brick_broadcast_invalid_symbol);
+					return false;
```
Comment (Gorphi) [inline catroid/src/org/catrobat/catroid/content/bricks/BroadcastReceiverBrick.java]: call to dismiss() missing?

File: catroid/src/org/catrobat/catroid/content/bricks/HideBrick.java [hunk 17]
```diff
@@ -105,8 +104,9 @@ public View getPrototypeView(Context context) {
 	}
 
 	@Override
+
```
Comment (Gorphi) [inline catroid/src/org/catrobat/catroid/content/bricks/HideBrick.java]: remove newline

**Comment by Gorphi** [comment]
remove all ActionFactory.sequence() calls from the bricks and replace them with actionFactoryObject.createSequence() to remove the import of the ActionFactory in the Bricks

File: catroid/src/org/catrobat/catroid/formulaeditor/InternFormulaKeyboardAdapter.java [hunk 18]
```diff
@@ -72,6 +72,7 @@
 				return buildNumber("9");
 
 			//FUNCTIONS:
+
```
Comment (Gorphi) [inline catroid/src/org/catrobat/catroid/formulaeditor/InternFormulaKeyboardAdapter.java]: remove newline

File: catroid/src/org/catrobat/catroid/io/StorageHandler.java [hunk 19]
```diff
@@ -368,6 +378,20 @@ private void setXstreamAliases() {
 		xstream.alias("userBrickElement", UserScriptDefinitionBrickElement.class);
 		xstream.alias("userBrickParameter", UserBrickParameter.class);
 
+		// Physics Script
+		// TODO physics change to normal script xml tag
```
Comment (Gorphi) [inline catroid/src/org/catrobat/catroid/io/StorageHandler.java]: remove TODO

File: catroid/src/org/catrobat/catroid/io/XStreamBrickConverter.java [hunk 20]
```diff
@@ -37,7 +37,8 @@
 public class XStreamBrickConverter extends ReflectionConverter {
 
 	private static final String TAG = XStreamBrickConverter.class.getSimpleName();
-	private static final String BRICKS_PACKAGE_NAME = "org.catrobat.catroid.content.bricks";
+	private static final String[] BRICKS_PACKAGE_NAMES = { "org.catrobat.catroid.content.bricks",
+			"org.catrobat.catroid.content.bricks.conditional", "org.catrobat.catroid.physics.content.bricks" };
```
Comment (Gorphi) [inline catroid/src/org/catrobat/catroid/io/XStreamBrickConverter.java]: remove org.catrobat.catroid.content.bricks.conditional

File: catroid/src/org/catrobat/catroid/io/XStreamToSupportCatrobatLanguageVersion098AndBefore.java [hunk 21]
```diff
@@ -491,19 +501,56 @@ private void initializeBrickInfoMap() {
 
 		brickInfo = new BrickInfo(ChooseCameraBrick.class.getSimpleName());
 		brickInfoMap.put("chooseCameraBrick", brickInfo);
+
+		// TODO[physics]
```
Comment (Gorphi) [inline catroid/src/org/catrobat/catroid/io/XStreamToSupportCatrobatLanguageVersion098AndBefore.java]: remove TODO

File: catroid/src/org/catrobat/catroid/physics/PhysicsBoundaryBox.java [hunk 22]
```diff
@@ -0,0 +1,91 @@
...
+		Body body = world.createBody(bodyDef);
+		body.createFixture(fixtureDef);
+		body.setUserData(identifier);
+
+		//		for (Fixture f : body.getFixtureList()) {
+		//			Filter filter = new Filter();
+		//			filter.categoryBits = 0;
+		//			filter.maskBits = 0;
+		//			f.setFilterData(filter);
+		//		}
```
Comment (Gorphi) [inline catroid/src/org/catrobat/catroid/physics/PhysicsBoundaryBox.java]: remove comments

**Comment by Gorphi** [comment]
ok to test

**Comment by fwinkelbauer** [comment]
Please also fix the format check by running the Android Studio Formatter like mentioned [here](https://github.com/Catrobat/Catroid/wiki/Creating-a-pull-request)

:+1: for doing reviews :D

File: catroid/src/org/catrobat/catroid/physics/PhysicsCollision.java [hunk 23]
```diff
@@ -0,0 +1,139 @@
...
+	private void registerContact(Sprite sprite1, Sprite sprite2) {
+		String key1 = generateKey(sprite1, sprite2);
+		String key2 = generateKey(sprite2, sprite1);
+		if (!physicsCollisionBroadcasts.containsKey(key1)) {
+			PhysicsCollisionBroadcast physicsCollisionBroadcast = new PhysicsCollisionBroadcast(sprite1.getName(), sprite2.getName());
+			physicsCollisionBroadcasts.put(key1, physicsCollisionBroadcast);
+			physicsCollisionBroadcasts.put(key2, physicsCollisionBroadcast);
+		}
+		physicsCollisionBroadcasts.get(key1).increaseContactCounter();
+		//Log.d(TAG, "registerContact:" + physicsCollisionBroadcasts.toString());
```
Comment (Gorphi) [inline catroid/src/org/catrobat/catroid/physics/PhysicsCollision.java]: remove comment

File: catroid/src/org/catrobat/catroid/physics/PhysicsCollision.java [hunk 24]
```diff
@@ -0,0 +1,139 @@
...
+	private void unregisterContact(Sprite sprite1, Sprite sprite2) {
+		String key1 = generateKey(sprite1, sprite2);
+		String key2 = generateKey(sprite2, sprite1);
+		if (physicsCollisionBroadcasts.containsKey(key1)) {
+			PhysicsCollisionBroadcast physicsCollisionBroadcast = physicsCollisionBroadcasts.get(key1);
+			physicsCollisionBroadcast.decreaseContactCounter();
+
+			if (physicsCollisionBroadcast.getContactCounter() == 0) {
+				physicsCollisionBroadcast.sendBroadcast();
+				//Log.d(TAG, "remove contact:" + physicsCollisionBroadcasts.toString());
```
Comment (Gorphi) [inline catroid/src/org/catrobat/catroid/physics/PhysicsCollision.java]: remove comment

File: catroid/src/org/catrobat/catroid/physics/PhysicsCollisionBroadcast.java [hunk 25]
```diff
@@ -0,0 +1,88 @@
...
+package org.catrobat.catroid.physics;
+
+import org.catrobat.catroid.ProjectManager;
+import org.catrobat.catroid.content.BroadcastEvent;
+import org.catrobat.catroid.content.Sprite;
+
+import java.util.List;
+
+public class PhysicsCollisionBroadcast {
+	//private static final String TAG = PhysicsCollisionBroadcast.class.getSimpleName();
```
Comment (Gorphi) [inline catroid/src/org/catrobat/catroid/physics/PhysicsCollisionBroadcast.java]: remove comment

File: catroid/src/org/catrobat/catroid/physics/PhysicsDebugSettings.java [hunk 26]
```diff
@@ -0,0 +1,40 @@
...
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+package org.catrobat.catroid.physics;
+
+public final class PhysicsDebugSettings {
+
+	private PhysicsDebugSettings() {
+	} // Make sure that utility classes (classes that contain only static methods or fields in their API) do not have a public constructor.
```
Comment (Gorphi) [inline catroid/src/org/catrobat/catroid/physics/PhysicsDebugSettings.java]: remove comment

File: catroid/src/org/catrobat/catroid/physics/PhysicsLook.java [hunk 27]
```diff
@@ -0,0 +1,353 @@
...
+				@Override
+				public boolean isTrue() {
+					return !isLookVisible();
+				}
+			};
+
+			transparencyCondition = new PhysicsObjectStateCondition() {
+				@Override
+				public boolean isTrue() {
+					//Log.d(TAG, "transparencyCondition:"+(alpha==0.0));
```
Comment (Gorphi) [inline catroid/src/org/catrobat/catroid/physics/PhysicsLook.java]: remove comment

File: catroid/src/org/catrobat/catroid/physics/PhysicsObject.java [hunk 28]
```diff
@@ -0,0 +1,468 @@
...
+		PhysicsObject.Type type = PhysicsObject.Type.NONE;
+		for (Brick brick : brickList) {
+			if (!(brick instanceof SetGravityBrick) && (brick.getRequiredResources() & Brick.PHYSIC) > 0) {
+				type = PhysicsObject.Type.DYNAMIC;
+				break;
+			}
+		}
+		setType(type);
+
+		// --
```
Comment (Gorphi) [inline catroid/src/org/catrobat/catroid/physics/PhysicsObject.java]: remove line

File: catroid/src/org/catrobat/catroid/physics/PhysicsWorldConverter.java [hunk 29]
```diff
@@ -0,0 +1,56 @@
...
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+package org.catrobat.catroid.physics;
+
+import com.badlogic.gdx.math.Vector2;
+
+public final class PhysicsWorldConverter {
+
+	private PhysicsWorldConverter() {
+	} // Make sure that utility classes (classes that contain only static methods or fields in their API) do not have a public constructor.
```
Comment (Gorphi) [inline catroid/src/org/catrobat/catroid/physics/PhysicsWorldConverter.java]: remove comment

File: catroid/src/org/catrobat/catroid/physics/content/ActionFactory.java [hunk 30]
```diff
@@ -0,0 +1,795 @@
...
+		return action;
+	}
+
+	public Action createPlaceAtAction(Sprite sprite, Formula x, Formula y) {
+		GlideToAction action = Actions.action(GlideToAction.class);
+		action.setPosition(x, y);
+		action.setDuration(0);
+		action.setInterpolation(null);
+		action.setSprite(sprite);
+		return action; //XXX: wrong action???
```
Comment (Gorphi) [inline catroid/src/org/catrobat/catroid/physics/content/ActionFactory.java]: remove comment

File: catroid/src/org/catrobat/catroid/physics/content/ActionPhysicsFactory.java [hunk 31]
```diff
@@ -0,0 +1,147 @@
...
+	// OVERRIDE
+	@Override
+	public Action createIfOnEdgeBounceAction(Sprite sprite) {
+		IfOnEdgeBouncePhysicsAction action = Actions.action(IfOnEdgeBouncePhysicsAction.class);
+		action.setSprite(sprite);
+		action.setPhysicsWorld(getPhysicsWorld());
+		return action;
+	}
+
+	// PHYSICS
```
Comment (Gorphi) [inline catroid/src/org/catrobat/catroid/physics/content/ActionPhysicsFactory.java]: remove comment

File: catroid/src/org/catrobat/catroid/physics/content/actions/GlideToPhysicsAction.java [hunk 32]
```diff
@@ -0,0 +1,52 @@
...
+import org.catrobat.catroid.physics.PhysicsLook;
+
+public class GlideToPhysicsAction extends GlideToAction {
+	private static final String TAG = GlideToPhysicsAction.class.getSimpleName();
+
+	private PhysicsLook physicsLook;
+
+	@Override
+	protected void begin() {
+		Log.d(TAG, "begin GlideToPhsicsAction");
```
Comment (Gorphi) [inline catroid/src/org/catrobat/catroid/physics/content/actions/GlideToPhysicsAction.java]: remove log

File: catroid/src/org/catrobat/catroid/physics/content/actions/GlideToPhysicsAction.java [hunk 33]
```diff
@@ -0,0 +1,52 @@
...
+		physicsLook.startGlide();
+		super.begin();
+	}
+
+	@Override
+	protected void end() {
+		Log.d(TAG, "end GlideToPhsicsAction");
```
Comment (Gorphi) [inline catroid/src/org/catrobat/catroid/physics/content/actions/GlideToPhysicsAction.java]: remove log

File: catroid/src/org/catrobat/catroid/physics/content/actions/IfOnEdgeBouncePhysicsAction.java [hunk 34]
```diff
@@ -0,0 +1,118 @@
...
+import com.badlogic.gdx.scenes.scene2d.actions.TemporalAction;
+
+import org.catrobat.catroid.ProjectManager;
+import org.catrobat.catroid.content.Sprite;
+import org.catrobat.catroid.physics.PhysicsBoundaryBox;
+import org.catrobat.catroid.physics.PhysicsWorld;
+
+public class IfOnEdgeBouncePhysicsAction extends TemporalAction {
+
+	//private static final String TAG = IfOnEdgeBouncePhysicsAction.class.getSimpleName();
```
Comment (Gorphi) [inline catroid/src/org/catrobat/catroid/physics/content/actions/IfOnEdgeBouncePhysicsAction.java]: remove comment

**Comment by Gorphi** [comment]
catroid/src/org/catrobat/catroid/ui/fragment/SpritesListFragment.java:copySprite() 

Check if it is necessary to use the updateCollisionBroadcastMessages(...) in here. If it isn't the visibility of that methode can be changed to private

**Comment by Gorphi** [comment]
GravityActionTest -> remove comments

**Comment by Gorphi** [comment]
PhysicsTestUtils -> remove comments

**Comment by Gorphi** [comment]
rename catroidTest/src/org/catrobat/catroid/uitest/content/brick/physics/SetGravityTest.java and catroidTest/src/org/catrobat/catroid/uitest/content/brick/physics/SetMassTest.java so that they contain Brick

File: catroid/src/org/catrobat/catroid/ProjectManager.java (hunk 2 above)
Comment (Gorphi) [inline catroid/src/org/catrobat/catroid/ProjectManager.java]: issue created https://jira.catrob.at/browse/CAT-1908?filter=11002

File: catroid/src/org/catrobat/catroid/common/DefaultProjectHandler.java (hunk 3 above)
Comment (Gorphi) [inline catroid/src/org/catrobat/catroid/common/DefaultProjectHandler.java]: done

**Comment by Gorphi** [comment]
Catroid Check: https://jenkins.catrob.at/view/LeeroyCatroid/job/CatroidCheck/68/
JUnit physics tests: https://jenkins.catrob.at/view/LeeroyCatroid/job/CatroidSinglePackageEmulatorTest/91/
UI physics tests: https://jenkins.catrob.at/view/LeeroyCatroid/job/CatroidSinglePackageEmulatorTest/97/

File: catroid/src/org/catrobat/catroid/content/bricks/Brick.java [hunk 35]
```diff
@@ -55,17 +55,20 @@
...
 
-	//use bitwise | for using multiple ressources in a brick
+	//use bitwise | for using multiple resources in a brick
 	int NO_RESOURCES = 0x0;
-	//	public static final int SOUND_MANAGER = 0x1;
+	//int SOUND_MANAGER = 0x1;
 	int TEXT_TO_SPEECH = 0x2;
 	int BLUETOOTH_LEGO_NXT = 0x4;
-	//	public static final int BLUETOOTH_ARDUINO = 0x8;
+	int PHYSIC = 0x8;
```
Comment (anddman) [inline catroid/src/org/catrobat/catroid/content/bricks/Brick.java]: rename to "PHYSICS"

Comment (Gorphi) [inline catroid/src/org/catrobat/catroid/content/bricks/Brick.java]: done

File: catroid/res/values/arrays.xml [hunk 36]
```diff
@@ -154,4 +154,12 @@
         <item>@string/phiro_sensor_bottom_right</item>
     </string-array>
 
+    <!-- Physics object type spinner -->
+    <string-array name="physics_object_types">
+        <item>@string/brick_set_physics_object_type_dynamic</item>
+        <item>@string/brick_set_physics_object_type_fixed</item>
+        <item>@string/brick_set_physics_object_type_no</item>
```
Comment (robertpainsi) [inline catroid/res/values/arrays.xml]: `brick_set_physics_object_type_none`

Comment (robertpainsi) [inline catroid/res/values/arrays.xml]: Or is it `type_no` now?

File: catroid/src/org/catrobat/catroid/common/LookData.java [hunk 37]
```diff
@@ -130,6 +130,9 @@ public Pixmap getPixmap() {
 				if (gdxRuntimeException.getMessage().startsWith("Couldn't load file:")) {
 					pixmap = new Pixmap(1, 1, Pixmap.Format.Alpha);
 				}
+			} catch (NullPointerException nullPointerException) {
+				Log.e(TAG, "gdx.files throws nullpointer");
```
Comment (robertpainsi) [inline catroid/src/org/catrobat/catroid/common/LookData.java]: log exception

File: catroid/src/org/catrobat/catroid/common/LookData.java [hunk 38]
```diff
@@ -130,6 +130,9 @@ public Pixmap getPixmap() {
...
+				pixmap = null;
```
Comment (robertpainsi) [inline catroid/src/org/catrobat/catroid/common/LookData.java]: Unnecessary -> Remove

File: catroid/res/values/arrays.xml (hunk 36 above)
Comment (Gorphi) [inline catroid/res/values/arrays.xml]: It is "no" now.

File: catroid/src/org/catrobat/catroid/common/LookData.java (hunk 37 above)
Comment (Gorphi) [inline catroid/src/org/catrobat/catroid/common/LookData.java]: done, (also for the Log.e a few lines above)

File: catroid/src/org/catrobat/catroid/common/LookData.java (hunk 38 above)
Comment (Gorphi) [inline catroid/src/org/catrobat/catroid/common/LookData.java]: done

**Comment by thmq** [comment]
ok to test

**Comment by Gorphi** [comment]
ok to test

**Comment by Gorphi** [comment]
ok to test

**Comment by Gorphi** [comment]
Physics unit tests: https://jenkins.catrob.at/view/LeeroyCatroid/job/CatroidSinglePackageEmulatorTest/104/
Physics UI tests: https://jenkins.catrob.at/view/LeeroyCatroid/job/CatroidSinglePackageEmulatorTest/105/

**Pull request merged**